leaderElection/src/NetNode.java

```diff
@@ -0,0 +1,95 @@
+import java.io.*;
+import java.util.HashMap;
+
+public class NetNode {
+
+    public static void main(String[] args) {
+        try {
+
+            String configPath = args.length > 0 ? args[0] : "../config.txt";
+            String nodeId = args.length > 1 ? args[1] : "-1";
+
+            Logger.setLocalNodeId(Integer.parseInt(nodeId));
+            Logger.Info("Init node......");
+
+            Node node = initNode(configPath, nodeId);
+
+            node.startMsgService();
+
+            StringBuilder tmp = new StringBuilder();
+            for (int nId : node.getNeighbors().keySet()) {
+                tmp.append(nId + "  ");
+            }
+            Logger.Info("Connecting Node: %s", tmp);
+
+            Logger.Info("Send Msg or Press [d/D] to disconnect by NodeId.");
+            while (true) {
+                BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+                String msg = reader.readLine();
+                if (msg == null || msg.isEmpty()) {
+                    continue;
+                }
+
+                if (msg.equalsIgnoreCase("D")) {
+                    Logger.Info("Input nodeId:");
+                    String ss = reader.readLine();
+                    if (ss == null || ss.isEmpty()) {
+                        continue;
+                    }
+                    node.disconnect(Integer.parseInt(ss));
+                } else {
+                    node.sendTestMsg(msg);
+                }
+            }
+        } catch (Exception e) {
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            e.printStackTrace(pw);
+            Logger.Error(sw.toString());
+        }
+    }
+
+
+    public static Node initNode(String configs, String nodeId) throws Exception {
+
+        HashMap<String, String> nodes = new HashMap();
+        try (BufferedReader br = new BufferedReader(new FileReader(configs))) {
+            String line;
+            Boolean addingNodes = false;
+            int lineNum = 0;
+            while ((line = br.readLine()) != null) {
+                lineNum++;
+                line = line.trim().toLowerCase();
+                if (isValidLine(line)) {
+                    String realLine = line.split("#")[0];
```

Comment:
Meaning

Reply:
reading config line by line.
Some unused variable here is removed by the later updates.